### DISCUSSION TITLE
Invalid XMP parsing error prevents to sign a PDF

### DISCUSSION THREAD

**Discussion by GaelC92** [MatthiasValvekens/pyHanko#473] [Q&A] [answered]
For context, this issue arose when trying to counter-sign an already signed pdf (which usually works) 

Analysis shows that the signature process fails when an invalid token appears in the XMP of the document to sign. 

Since the document is already signed once, it is not clear wether this should work or if the existing signature is fully valid – however, it seems that if the document was signed once, it should be possible to sing it again.

It appears that the parsing is required in order to update the XMP with Producer and LastModified data – could this behavior possibly be disabled for such cases ? 

Here is the related call stack at the point it fails : 
![image](https://github.com/user-attachments/assets/a099d7d3-53ce-4870-a7f1-1f8bdf4375cc)

**Environment (please complete the following information):**
 - OS: Mac 0S 14.5 
 - Version 0.25 (Python 3.12.4 venv)

Exemple of such invalid XMP : 
xmp = (b'\n<x:xmpmeta x:xmptk="3.1-701" xmlns:x="adobe:ns:meta/">\n<rdf:RDF '
        b'xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">\n<rdf:Description rdf:about="" '
        b'xmlns:pdf="http://ns.adobe.com/pdf/1.3/">\n    '
        b'<pdf:Producer>\xc3\xbe\xc3\xbf\x00M\x00i\x00c\x00r\x00o\x00s\x00o\x00f\x00t\x00\xc2\xae\x00 '
        b'\x00W\x00o\x00r\x00d\x00 \x002\x000\x001\x006</pdf:Producer>\n</rdf:Description>\n<rdf:Description '
        b'rdf:about="" xmlns:xmp="http://ns.adobe.com/xap/1.0/">\n    <xmp:CreatorTool>Microsoft\xc2\xae Word '
        b'2016</xmp:CreatorTool>\n    <xmp:CreateDate>2018-12-13T10:30:29+01:00</xmp:CreateDate>\n    '
        b'<xmp:ModifyDate>2024-08-01T11:07:19+02:00</xmp:ModifyDate>\n<xmp:MetadataDate>2024-08-01T11:07:19+02:00'
        b'</xmp:MetadataDate></rdf:Description>\n<rdf:Description rdf:about="" '
        b'xmlns:xmpMM="http://ns.adobe.com/xap/1.0/mm/">\n    '
        b'<xmpMM:DocumentID>uuid:37102f1d-b6ad-9d2c-8d34-9d063a60c7a1</xmpMM:DocumentID>\n    '
        b'<xmpMM:InstanceID>uuid:4f8e5be0-aeae-bdb7-34b9-b09e4d90e31f</xmpMM:InstanceID>\n</rdf:Description>\n</rdf'
        b':RDF>\n</x:xmpmeta>                                                                                         '
        b'          \n                                                                                                '
        b'   \n                                                                                                   \n  '
        b'                                                                                                 \n         '
        b'                                                                                          \n                '
        b'                                                                                   \n                       '
        b'                                                                            \n                              '
        b'                                                                     \n                                     '
        b'                                                              \n                                            '
        b'                                                       \n                                                   '
        b'                                                \n                                                          '
        b'                                         \n                                                                 '
        b'                                  \n                                                                        '
        b'                           \n                                                                               '
        b'                    \n                                                                                      '
        b'             \n                                                                                             '
        b'      \n                                                                                                   '
        b'\n                                                                                                   \n     '
        b'                                                                                              \n<?xpacket '
        b'end="w"?>')

**Comment by MatthiasValvekens** [comment, marked as answer]
Hi, there's indeed a way to disable that behaviour if you don't need it--I don't remember off the top of my head whether it has flag or not (don't have my laptop here to check), but overriding the `_update_meta` (sp?) method on the PDF writer to be a no-op should have the desired result. There are some tests that use that trick, mostly to generate invalid files and the like.

**Comment by GaelC92** [comment]
Thanks Matthias. 
Sorry and my apologies for the delayed response. I will check and see if that fixes the issue.

**Reply by GaelC92** [reply]
Tested, and it works : 

` w._update_meta = lambda:  None
`

However, it may be worth it to expose a method or property allow this bypass in a less hacky way.
